### ISSUE TITLE
Multiple tree in view causing issue 

### ISSUE THREAD

**Issue by sarahm7** [Zicrael/ngx-tree-dnd#44]
Need help when i use multiple instance of the component the + event is not triggering on the correct component

See example codes here:
https://stackblitz.com/edit/ngx-tree-dnd-pj9zw6?file=app%2Ftree%2Ftree.component.ts

Even if i clicked the + icon on the first instance, the last instance of the component gets updated instead

![image](https://user-images.githubusercontent.com/27810542/53955608-6ddf1a80-4114-11e9-8d8d-726092714dfe.png)

**Comment by Zicrael** [comment]
@sarahm7 ,

Yes, I think it required a lot of changes. I will try fix repository after I finish another new pachage ( vue-tree-dnd) and take example of code from new package,  because I write it clear and smart (have much more expirience since this repo) . 
Thank you for download it,  I am sorry for this bugs.

**Comment by sarahm7** [comment]
@Zicrael thanks for the reply! i'm trying to fix it in my end but i'm not sure if the approach is good for you. I placed a rootId on the tree model so i can keep track which instance it is. im testing it now and if its good already i'll create a pull request later.

**Comment by NoureddineRmila** [comment]
@Zicrael  we can't use this package for drag and drop an element between two trees?

**Comment by Zicrael** [comment]
@RmilaWeb  at the current version - no:( But I hope I will fix it in next versions.

**Comment by NoureddineRmila** [comment]
> @RmilaWeb at the current version - no:( But I hope I will fix it in next versions.

thanks for the reply! Can you tell me when the next version? Cause need this future.

**Comment by hirenkukadiya** [comment]
@sarahm7 can you make multiple tree?
I placed a rootId on the tree model so i can keep track which instance it is. im testing it now and if its good already i'll create a pull request later.
If its work then please share your experience.. 
and how it work.?

**Comment by sarahm7** [comment]
yes i was able to do it after modifying the code. although i can't do a pull request at the moment since i'm still testing it. basically you can use it like this:

<app-tree [rootId]="1"></app-tree>
<app-tree [rootId]="2"></app-tree>
<app-tree [rootId]="3"></app-tree>
<app-tree [rootId]="4"></app-tree>

![image](https://user-images.githubusercontent.com/27810542/54739383-dc32db00-4bf2-11e9-8c2c-319734e5a1b0.png)

so the singleton tree-service can handle the event properly based on the rootId. i'm not sure if its the best solution though.

**Comment by Zicrael** [comment]
@sarahm7 , @hirenkukadiya  Thank you for support this. I don`t have so much time now for package, but I can make you collabarators. Also I almost finish new plugin for vue. Better, faster, a lot of functionality, I share it with you guys.
